internal/component_model.go

```diff
@@ -111,12 +117,85 @@ func (m *componentResourceModel) toRemoteInput() sbmgmt.ComponentInput {
 	// using the ordering of the json...
 	schema := sortComponentFields(raw)
 
-	return sbmgmt.ComponentInput{
-		Name:               m.Name.ValueString(),
-		Schema:             schema,
-		IsNestable:         m.IsNestable.ValueBoolPointer(),
-		IsRoot:             m.IsRoot.ValueBoolPointer(),
-		ComponentGroupUuid: ref(asUUID(m.ComponentGroupUUID)),
+	componentGroupUuid := uuid.Must(uuid.FromString(m.ComponentGroupUUID.ValueString()))
+
+	return sbmgmt.CreateComponentJSONRequestBody{
```

Comment:
Perhaps we should have named the object in openapi to generate `sbmfmt.ComponentCreateInput{}`